pipeline/pump.go

```diff
@@ -0,0 +1,43 @@
+package pipeline
+
+import (
+	"context"
+
+	"github.com/dogmatiq/dodeca/logging"
+	"github.com/dogmatiq/marshalkit"
+	"golang.org/x/sync/errgroup"
+)
+
+// Pump obtains sessions a source and feeds them to a sink.
```

Comment:
Does it obtain a session source or does it obtain a source for sessions?

Reply:
Should say "from a source", thx :)